airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py

```diff
@@ -30,7 +30,8 @@ class SparkKubernetesOperator(BaseOperator):
         For more detail about Spark Application Object have a look at the reference:
         https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/v1beta2-1.1.0-2.4.5/docs/api-docs.md#sparkapplication
 
-    :param application_file: filepath to kubernetes custom_resource_definition of sparkApplication
+    :param application_file: either 1) a file ending in '.yaml', '.yml', or '.json' that will be converted to the file's string contents by the templating, or
+                                    2) a JSON string defining kubernetes custom_resource_definition of sparkApplication
```

Comment:
This is restructuredText, so I don’t think the indentation and `1)` will do what you want in the documentation. I would write this as prose, e.g.

```
Defines Kubernetes `custom_resource_definition` of `sparkApplication. Either a path to a ``.json`` file, or a JSON string.
```

(Does YAML make sense for this field? I’m not sure.)

Reply:
Thanks!  I've updated.  Per https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CustomObjectsApi.md#create_namespaced_custom_object,which is what's ultimately called, I don't think YAML is supported, just JSON

Reply:
I don’t know why the YAML options are there, let’s keep them there for others with more knowledge on this topic…